util/syspolicy/setting/origin.go

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package setting
+
+import (
+	"fmt"
+
+	jsonv2 "github.com/go-json-experiment/json"
+	"github.com/go-json-experiment/json/jsontext"
+)
+
+// Origin describes where a policy or a policy setting is configured.
+type Origin struct {
+	data settingOrigin
+}
+
+// settingOrigin is the marshallable data of an [Origin].
+type settingOrigin struct {
+	Name  string `json:",omitzero"`
```

Comment:
The `omitzero` tag can be used with [jsonv2](https://github.com/go-json-experiment/json), and will likely be backported to `encoding/json`.

The meaning of `omitempty` in `jsonv2` differs somewhat from its use in `json`. Even though `omitempty` and `omitzero` produce the same JSON for string fields, I think _"omit Name if it has its zero value"_ better reflects my intent than _"omit Name if it would be marshaled to an empty JSON value"_.